tests/foreign_object/tests.py

```diff
@@ -365,6 +365,10 @@ def test_many_to_many_related_query_name(self):
         with self.assertRaises(FieldError):
             Article.objects.filter(ideas__name="idea1")
 
+    def test_ticket_24352(self):
```

Comment:
You can use a more descriptive name like `test_many_related_manager_str`. I think a better place for this test would be `m2m_regress`. Currently, your test is in a class called `MultiColumnFKTests` which isn't relevant to this change.